sycl/doc/OptionalDeviceFeatures.md

```diff
@@ -629,11 +629,122 @@ example, if a new device is released before there is a new DPC++ release.  In
 fact, the DPC++ driver supports a command line option which allows the user
 to select an alternate configuration file.
 
-**TODO**: More information will be inserted here when we merge
-[this separate PR][7] into this design document.
+**TODO**: 
+* The names of the devices in the configuration file.
+* Location of the default device configuration file
+
+#### New features in clang compilation driver and tools
+
+NOTE: the term *device binary image* or *image* is used to refer to a device
+code form consumable by the SYCL runtime. Earlier device code forms are referred
+to as *device code module* or *device IR module*. In case of AOT, device binary
+image is a natively compiled binary, and IR module - either a SPIR-V or LLVMIR
+bitcode module.
+
+##### Overview
+After the `sycl-post-link` performs necessary aspect usage analysis and splits
+the incoming monolythic device code module into pieces - smaller device code
+modules - it outputs a file table as a result. Each row in the table corresponds
+to an individual output module, and each element of a row is a name of a file
+containing necessary information about the module, such as the code itself, its
+properties.
+
+At the action graph building stage for each requested AOT compilation target -
+SPIR-V-based (such as Gen targets) and/or non-SPIR-V-based (such as PTX) - the
+driver adds an `aspect-filter` action which filters out input file table rows
+with device code modules using features unsupported on current target. Then the
+output table goes as input into the AOT stage, and the prior filtering
+guarantees that the AOT compiler will not encounter device code it can't
+compile. In the extreme case when all device code
+modules use unsupported aspects, the input file table will be empty. The picture
+below illustrates the action graph built by the clang driver along with file
+lists and tables generated and consumed by various nodes of the graph. The
+example set of targets used for the illustration is 4 targets
+- spir64 (runtime JITted SPIR-V)
+- AOT targets
+    - non-SPIR-V based
+        - nvptx (PTX)
+    - SPIR-V based
+        - gen_11 (Intel Gen)
+        - x86_64_avx512 (AVX512)
+<br>
+<br>
+
+![Device SPIRV translation and AOT compilation](images/DeviceLinkAOTAndWrap.svg)
```

Comment:
I think it might be helpful to change the wording in the boxes in this SVG file that are currently labeled "Aspect filter" to show the real tool name and its command line.  Therefore, a typical box would have text like:

```
sycl-feature-filter -target=ptx64
```

This will help connect the diagram to the tool description below.

Reply:
> I think it might be helpful to change the wording in the boxes in this SVG file that are currently labeled "Aspect filter" to show the real tool name and its command line.

OK

> sycl-feature-filter

the tool really filters based on aspects, so I'd stay with sycl-aspect-filter, it also gives a connection to the spec